src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -385,58 +384,87 @@
-        profile_type='${genomicDataFilter.profileType}'
-        <foreach item="dataFilterValue" collection="genomicDataFilter.values" open=" AND ((" separator=") OR (" close="))">
-            <trim prefix="" prefixOverrides="AND">
-                <if test="dataFilterValue.value eq 'NA'">
-                    AND
-                    <include refid="isAttributeValueNA">
-                        <property name="attribute_value" value="alteration_value"/>
-                    </include>
-                </if>
-                <if test="dataFilterValue.start != null and dataFilterValue.end == null">
-                    AND match(alteration_value, '^>?=?[-+]?[0-9]*[.,]?[0-9]+$')
-                </if>
-                <if test="dataFilterValue.start == null and dataFilterValue.end != null">
-                    AND match(alteration_value, '^&lt;?=?[-+]?[0-9]*[.,]?[0-9]+$')
-                </if>
-                <if test="dataFilterValue.start != null and dataFilterValue.end != null">
-                    AND match(alteration_value, '^[-+]?[0-9]*[.,]?[0-9]+$')
-                </if>
-                <if test="dataFilterValue.start != null or dataFilterValue.end != null">
-                    <choose>
-                        <when test="dataFilterValue.start == dataFilterValue.end">
-                            AND abs(
-                            minus(
-                            <include refid="castStringValueToFloat">
-                                <property name="attribute_value" value="alteration_value"/>
-                            </include>,
-                            ${dataFilterValue.start}
-                            )
-                            ) &lt; exp(-11)
-                        </when>
-                        <otherwise>
-                            <if test="dataFilterValue.start != null">
-                                AND
-                                <include refid="castStringValueToFloat">
-                                    <property name="attribute_value" value="alteration_value"/>
-                                </include> &gt; ${dataFilterValue.start}
-                            </if>
-                            <if test="dataFilterValue.end != null">
-                                AND
+    <sql id="numericalGenomicDataFilter">
+        <!-- check if 'NA' is selected -->
+        <bind name="containsNA" value="false" />
+        <bind name="containsNumericalValue" value="false" />
+        <foreach item="dataFilterValue" collection="genomicDataFilter.values">
+            <choose>
+                <when test="dataFilterValue.value == 'NA'">
+                    <bind name="containsNA" value="true" />
+                </when>
+                <otherwise>
+                    <bind name="containsNumericalValue" value="true" />
+                </otherwise>
+            </choose>
+        </foreach>
+        <if test="containsNA">
```

Comment:
This here gives us all alterations where value is null in all the studies being queried.  It encompasses both case where value is actually null and where a sample is not profiled.  it's important to note that this list is not samples so much as it is a list of of alterations (?)

Reply:
note to self:  because of the LEFT join, if a sample has no alteration, it will appear in this list with null values for the joined fields and so will be counted as NA

Reply:
@fuzhaoyuan lets improve naming/commenting.  e.g. containsNA would be clearer "filteringForNA"